src/gil.rs

```diff
@@ -208,6 +213,14 @@ impl GILGuard {
             }
         }
 
+        // This is necessary to protect against Python::with_gil being called after the Python
+        // interpreter has started shutting down.
+        assert_eq!(
```

Comment:
Could we somehow move is behind the `if gil_is_acquired()` check to avoid paying this cost in the no-op case of nested calls to `with_gil`?

Reply:
Ah sorry, it already is behind the check because the check itself is duplicated. Sorry for noise.